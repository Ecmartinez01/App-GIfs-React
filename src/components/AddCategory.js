import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    const [texto , setTexto ] = useState("")

    const encuentraTexto = (e)=>{
        setTexto(e.target.value)
    }
    const submit = (e)=>{
        e.preventDefault()
        console.log("Submit")
        if (texto.trim().length > 2) {
            setCategory(cats => [texto,...cats])
            setTexto('')
        }
    }
    return (
        <form onSubmit={submit} >
            <p> {texto}</p>
            <input
            value={texto}
            type="text"
            onChange={encuentraTexto}
            />
        </form>
        
    )
}
AddCategory.propTypes ={
    setCategory : PropTypes.func.isRequired       
}