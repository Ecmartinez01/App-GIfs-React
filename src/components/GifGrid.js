import React /*{ useEffect, useState }*/ from 'react';
import { useFetchGif } from '../hooks/useFetchGifs';
// import { getGif } from '../helpers/getGif';
import { GifGridItem } from './GifGridItem';
export const GifGrid =({categoria})=>{

      const {data,loading} = useFetchGif(categoria)

    
    return(<>
        <h3 className='animate__animated animate__flip animate__delay-0.5s '>{categoria}</h3>
        {loading && <p className='animate__animated animate__bounceOut animate__delay-1s '>Cargando</p>}
        <div className='card-grid'>
            {data.map(gif => (
               <GifGridItem 
               key={gif.id}
               {...gif}
               /> 
                
            ))}
      
        </div>
    </>
    )
}