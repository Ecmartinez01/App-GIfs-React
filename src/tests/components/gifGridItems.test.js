import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';
import React from 'react';

describe('Pruebas con gif grid Items', () => {


    const url = "https://localhost/algo.jpg";
    const title = "hola mundo";
   const  wrapper = shallow(<GifGridItem url={url} title={title} />)
 
    it('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Debe de tener un parrafo con el titulo', () => {
      const p = wrapper.find('p').text().trim()
      expect(p).toBe(title)
    })

    it('Debe de tener la imagen igual a la url y alt', () => {

      const img = wrapper.find('img')
      expect(img.prop("src")).toBe(url)
      expect(img.prop("alt")).toBe(title)
      })

    it('Bebe de contener animate__bounceInUps', () => {
        const div = wrapper.find("div")
        expect(div.prop("className")).toContain("animate__bounceInUps")
    })
      
    
})
