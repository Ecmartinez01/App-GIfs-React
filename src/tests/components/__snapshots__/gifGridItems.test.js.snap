// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con gif grid Items debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceInUps animate__delay-1s "
>
  <img
    alt="hola mundo"
    src="https://localhost/algo.jpg"
  />
  <p>
    hola mundo
  </p>
</div>
`;
